packages/ia-components/sandbox/details/DetailsDownloadOptions.jsx

```diff
@@ -71,26 +77,31 @@ export default class DetailsDownloadOptions extends React.Component {
             >
               {formats('format', k).downloadable}
               {' '}
-              <I18nIcon className="iconochive-download" en="download"/>
+              <I18nIcon className="iconochive-download" en="download" />
             </AnchorDownload>
+            {(typeof DwebArchive === 'undefined' || !DwebArchive.mirror || (k !== 'Epub')) ? null : (
+              <a href={`/epubreader/index.html?bookPath=/download/${this.props.identifier}/${downloadableFilesDict[k][0].metadata.name}`} title="Read online">
+                <I18nIcon className="iconochive-eye" en="Read online" />
+              </a>
+            )}
           </div>
         ))}
         <div className="show-all">
-          {(this.props.disconnected) ? null :
+          {(this.props.disconnected) ? null : (
             <div className="pull-right">
               <a className="boxy-ttl hover-badge" href={compressURL}>
-                <I18nIcon className="iconochive-download" en="download"/>
-                {' '}{filesCount}{' '}{I18nStr("Files")}
+                <I18nIcon className="iconochive-download" en="download" />
+                {' '}{filesCount}{' '}{I18nStr('Files')}
               </a>
-              <br/>
+              <br />
               <a className="boxy-ttl hover-badge" href={compressAllURL}>
-                <I18nSpan className="iconochive-download" en="download"/>
-                {originalFilesCount}{' '}{I18nStr("Original")}
+                <I18nSpan className="iconochive-download" en="download" />
+                {originalFilesCount}{' '}{I18nStr('Original')}
               </a>
-              <br/>
+              <br />
             </div>
-          }
-          <AnchorDownload className="boxy-ttl" identifier={this.props.identifier} disconnected={this.props.disconnected}><I18nSpan en="SHOW ALL"/></AnchorDownload>
+          )}
+          <AnchorDownload className="boxy-ttl" identifier={this.props.identifier} disconnected={this.props.disconnected}><I18nSpan en="SHOW ALL" /></AnchorDownload>
           <br clear="all" className="clearfix" />
```

Comment:
ahhh clearfix.  do we need?

Reply:
I don't know, wherever possible I copy the html that comes out of the existing petabox, and its there in the petabox code (search for ` <br clear="all" class="clearfix"/>` in https://archive.org/details/commute for example.  I periodically review the HTML my code generates against petabox html, but try not to get ahead of petabox code as then I get bitten by changes to archive.css that (quite reasonably) aren't tested against my code